### PR TITLE
Rewrite pystack logic & Update auto-reload

### LINKED ISSUE #3502: Double start of `code.py`
I'm experiencing a double start of "code.py" with beta 1 when powering the board from USB.

My APA cycles through a rainbow in code.py and I can see it goes from red to blue then turns off and then starts again at red.

This doesn't happen in beta 0 with the same code, and doesn't happen if I power the board from external 5V with no USB connected.

It also doesn't happen from a soft reset, so I suspect it's something to do with the mounting of the USB.

Update: Monitoring UART0 during the boot process shows nothing weird and no ESP32 specific restarts.
```ESP-ROM:esp32s2-rc4-20191025
Build:Oct 25 2019
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:2
load:0x3ffe6100,len:0x8
load:0x3ffe6108,len:0x18a0
load:0x4004c000,len:0x14b0
load:0x40050000,len:0x2134
entry 0x4004c2cc```

### PR DESCRIPTION
**Pull request by microdev1**
Two unrelated fixes:

- Rewrite `allocate_pystack` logic:
  - Saves space, fixes translation string, allow pystack to be set only when `SAFE_MODE_NONE` and remove upper size limit.
- Don't call `reload_initiate` if already initiated:
  - Stumbled upon this while trying to investigate #3502.

### CODE REVIEW

File: supervisor/shared/reload.c [hunk 1]
```diff
@@ -82,21 +82,21 @@ inline bool autoreload_is_enabled() {
...
+    // Initiate a reload of the VM immediately. Later code will pause to
+    // wait for the autoreload to become ready. Doing the VM exit
+    // immediately is clearer for the user.
+    if (!reload_initiated) {
         reload_initiate(RUN_REASON_AUTO_RELOAD);
     }
 }
 
 bool autoreload_ready() {
-    if (last_autoreload_trigger == 0 || autoreload_suspended != 0) {
```
Comment (tannewt) [inline supervisor/shared/reload.c:99]: Since `autoreload_suspended` is a bitmask, not a bool. Please do comparisons to zero. I can never remember the truthiness of ints. Doing an explicit comparison is clearer.

File: supervisor/shared/reload.c [hunk 2]
```diff
@@ -82,15 +82,15 @@ inline bool autoreload_is_enabled() {
 }
 
 void autoreload_trigger() {
-    if (autoreload_enabled & !autoreload_suspended) {
-        last_autoreload_trigger = supervisor_ticks_ms32();
-        // Guard against the rare time that ticks is 0;
```
Comment (tannewt) [inline supervisor/shared/reload.c:87]: Why did you remove this bit?

Comment (microdev1) [reply supervisor/shared/reload.c:87]: You mean "Guard against the rare time that ticks is 0"?
Do we want to trigger auto reload when ticks is 0? Will there even be such a case?

**Review by tannewt — APPROVED** [review]
Thank you!
